models/iaf_psc_delta.cpp

```diff
@@ -311,7 +311,7 @@ nest::iaf_psc_delta::update( Time const& origin, const long from, const long to
     }
 
     // threshold crossing
-    if ( S_.y3_ >= P_.V_th_ )
+    if ( S_.y3_ >= P_.V_th_ and S_.r_ == 0 )
```

Comment:
@akorgor Would you please explain in what sense this is a fix? A fix to what?

Reply:
If I understand the code correctly, without the additional condition `S_.r_ == 0`, the neuron can spike during the refractory period, which it should not be able to do.

Reply:
I understand. Initially, I thought this update was primarily about enhancing readability, similar to the other revisions. The term 'fix' implies that this commit addressed a specific 'bug' or problem, whereas it actually seems to have 'clarified' the firing conditions. The important thing here is that there was no actual 'bug' involved.

Reply:
Apologies, I had a thought of error. Looking at it again I realized that since the membrane potential is clamped to `V_reset` during the refractory period it should be safe without the extra condition. I indeed thought there was a bug. However, I think, that for the adaptive neuron we have to move the update of the adaptive threshold inside the not-refractory brackets in the if-clause because otherwise the condition could become true for with a changed adaptive threshold potential. This is where I originally stumbled over this condition.

Reply:
Yes, that is a good observation